src/server/transaction.cc

```diff
@@ -647,30 +648,12 @@ void Transaction::UnlockMulti() {
 }
 
 void Transaction::SetMultiUniqueShardCount() {
-  uint32_t prev = run_count_.fetch_add(shard_data_.size(), memory_order_relaxed);
-  DCHECK_EQ(prev, 0u);
-
-  std::atomic<uint32_t> unique_shard_cnt = 0;
-
-  auto update_shard_cnd = [&] {
-    EngineShard* shard = EngineShard::tlocal();
-    auto journal = shard->journal();
-
-    if (journal != nullptr) {
-      TxId last_tx = journal->GetLastTxId();
-      if (last_tx == txid_) {
-        unique_shard_cnt.fetch_add(1, std::memory_order_relaxed);
-      }
+  unique_shard_cnt_ = 0;
```

Comment:
can you explain what's the semantics of this function and why override `unique_shard_cnt_` for that?
i.e. how this is different from a "regular" `unique_shard_cnt_` ?

Also, even we need it, why do you need to store it inside unique_shard_cnt_ and not just return from this function and use this function when you need to know the value?

Reply:
Overriding the same variable with different semantics makes me nervous. This is why I am asking these questions.

Reply:
I updated the comments in the header 
It made sense to me to update the unique_shard_cnt_ var to hold the shard count in the whole transaction but as it is only updated in case of journal write I updated in the new commit to have this value returned by the function